### PR TITLE
intel hex file merging/manipulations

### PR DESCRIPTION
**Pull request by drahnr**

### CODE REVIEW

File: src/stitch.rs [hunk 1]
```diff
@@ -97,6 +102,20 @@ where T: Clone,
     Ok((sorted_vec, offset_sorted))
 }
 
+fn check_file_format(name: &Path) -> Result<FileFormat> {
+    let ext = match name.extension() {
+        Some(e) => e.to_str().unwrap(),
+        None => return Err(ScalpelError::UnknownFileFormat.context(format!("No extension found") ).into()),
+    };
+
+    match ext {
```
Comment (drahnr) [inline src/stitch.rs:162]: Maybe impl the `trait FromStr` directly for `FileFormat` or use a impl of `from_file_extension`?

File: src/stitch.rs [hunk 2]
```diff
@@ -121,5 +140,29 @@ mod test {
...
+    }
+
+    #[test]
+    fn test_no_ext() {
+        let name = PathBuf::from("tmp/test_bytes");
+        let ext = check_file_format(name.as_ref());
+
+        assert!(ext.is_err());
+    }
+
```
Comment (drahnr) [inline src/stitch.rs:291]: :+1:

File: src/stitch.rs (hunk 1 above)
Comment (drahnr) [reply src/stitch.rs:162]: Also, for the intermediate future, see #4

**Review by drahnr — COMMENTED** [review]
Can not approve, I created the PR :) but looks good to me

File: src/stitch.rs (hunk 1 above)
Comment (KuabeM) [reply src/stitch.rs:162]: I think I'll go for the `FromStr`, but if #4 is handling this stuff anyways, I'd leave it for now?

**Comment by KuabeM** [comment]
Still need some changes, the `write_hex_file` is never used. Should I add a flag for it or should we use the same output depending on the input format?

**Comment by KuabeM** [comment]
For `replace` and `stitch`, files can be in hex or binary format now. The `cut` part needs major modifications since it directly writes the cut out bytes to a file.

**Comment by drahnr** [comment]
> Should I add a flag for it or should we use the same output depending on the input format?

I'd say use the flag you introduced, but make it optional. In the case it is not passed to as argument, try to deduce from the output filename which file type it is going to be (maybe warn if it does not match the explicitly given one?). If none is inferable, bail and error out? But that's just an idea.

**Comment by KuabeM** [comment]
I went for the easy way, I just default to binary if no flag is given... Maybe together with #4?

Now everything should be fine, but maybe we should add more tests :)

**Comment by KuabeM** [comment]
I did a `git rebase origin/master` and `git push --force` to update to `master`

**Comment by drahnr** [comment]
If you say this is merge ready, I'd release an initial alpha

**Comment by KuabeM** [comment]
Yep should be goot to go

**Review by KuabeM — APPROVED** [review]

**Pull request merged**